components/apimgt/org.wso2.carbon.apimgt.ballerina.gateway/src/main/ballerina/gateway/api_gateway_listener.bal

```diff
@@ -183,7 +183,15 @@ function initiateGatewayConfigurations(EndpointConfiguration config) {
     }
     config.host = getConfigValue(LISTENER_CONF_INSTANCE_ID, LISTENER_CONF_HOST,"localhost");
     intitateKeyManagerConfigurations();
+    initiateThrottleConfigs();
+}
 
+function initiateThrottleConfigs() {
+    ThrottleConf config;
+    config.enabledHeaderConditions = getConfigBooleanValue(THROTTLE_CONF_INSTANCE_ID, "enabledHeaderConditions", false);
+    config.enabledJWTClaimConditions = getConfigBooleanValue(THROTTLE_CONF_INSTANCE_ID, "enabledJWTClaimConditions", false);
+    config.enabledQueryParamConditions = getConfigBooleanValue(THROTTLE_CONF_INSTANCE_ID, "enabledQueryParamConditions", false);
+    getGatewayConfInstance().setThrottleConf(config);
 }
```

Comment:
We can Initialize it like below as well. This is another option only
ThrottleConf config = {enabledHeaderConditions: getConfigBooleanValue(THROTTLE_CONF_INSTANCE_ID, "enabledHeaderConditions", false),
enabledJWTClaimConditions: getConfigBooleanValue(THROTTLE_CONF_INSTANCE_ID, "enabledJWTClaimConditions", false),
enabledQueryParamConditions : getConfigBooleanValue(THROTTLE_CONF_INSTANCE_ID, "enabledQueryParamConditions", false)
}